.github/e2e-tests-olm/action.yaml

```diff
@@ -67,6 +67,7 @@ runs:
       shell: bash
       run: |
         # TODO: build a kubernetes-specific bundle to avoid doing this
+        kubectl label nodes -l node-role.kubernetes.io/master=="" node-role.kubernetes.io/infra=""
```

Comment:
I am wondering if this can be by setting the labels in kind/config.yaml itself (need to try this one out). I also feel we shouldn't stop this PR from merging since it accomplishes the same (no 🚲 shedding)

Reply:
Yes, we need this in because we need to ensure that the operators run on infra-nodes.
@jan--f I agree with @JoaoBraveCoding it would be better to have the Jira issue mentioned in the commit 👼 

Also there, doesn't seem like a better way to label the node as infra using kind config.

see: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/ --node-labels mapStringString 
> Labels must be key=value pairs separated by ','. Labels in the 'kubernetes.io' namespace must begin with an allowed prefix ('kubelet.kubernetes.io', 'node.kubernetes.io')  

This needs to be documented otherwise when we add a test (may be in a followup PR) to validate the node where the pod is deployed is infra, that test will fail when run locally.

Reply:
If we are going to have a test to assert that the pods are running on nodes with the infra label then this makes absolute sense to have in the CI and I completely redact my comment. But since there wasn't a test this just seemed weird. (I never intended to 🚲 shed this, I was just curious 😅)

Reply:
>  (I never intended to 🚲 shed this, I was just curious 😅) 

sorry, I didn't mean you were bike shedding ( I did see the LGTM ) 👼  rather I was only pointing that even if this can be achieved through kind config (which at present isn't possible) it shouldn't stop the PR from merging since both methods achieve the same outcome.

Reply:
@jan--f could you please confirm that kind does create a node with `node-role.kubernetes.io/master==""` set

Testing this locally, I didn't find a 
```
❯ kc get nodes --show-labels
NAME                STATUS   ROLES           AGE   VERSION   LABELS
mso-control-plane   Ready    control-plane   48s   v1.24.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=mso-control-plane,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
mso-worker          Ready    <none>          27s   v1.24.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=mso-worker,kubernetes.io/os=linux
mso-worker2         Ready    <none>          27s   v1.24.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=mso-worker2,kubernetes.io/os=linux
```

Reply:
> @jan--f could you please confirm that kind does create a node with `node-role.kubernetes.io/master==""` set

Good catch, this should be `node-role.kubernetes.io/control-plane==""`